### PR TITLE
MAINT: Update conftest to check PyErr_Occurred

### PR DESCRIPTION
**Pull request by hoodmane**
I think there are some bugs where pyodide leaves the python error flag in an inconsistent state. These bugs invoke unspecified behavior in CPython. They often go unnoticed because the error flag is frequently cleared or overwritten. However, under the right circumstances they can show up, often due to changes that don't directly touch the code causing the problem. For instance, bug #1110 was exposed in PR #1098 because I added extra error checking code, even though bug #1110 is caused by problems in `python2js.c` and `python2js_buffer.c` which were both unchanged in #1098. 

I have seen similar symptoms to the ones caused by #1110 without hitting that particular codepath, so I strongly suspect there are still other similar bugs in the code. This will help by double checking whether the Python error indicator is in an invalid state at the end of each `selenium.run_js` call.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -108,7 +114,18 @@ def run_js(self, code):
             Error.stackTraceLimit = Infinity;
             let run = () => { %s }
             try {
-                return [0, run()]
+                let result = run();
+                if(pyodide._module._PyErr_Occurred){
```
[the remark this answers is not in the document]
Comment (hoodmane) [reply conftest.py]: Yes.

**Comment by hoodmane** [comment]
@dalcde Every test fails here, the errors are `TypeError: Cannot read property '_PyErr_Occurred' of undefined` from `pyodide._module._PyErr_Occurred`. So the problem is `pyodide._module` is undefined. Why would that be?? Maybe if we changed `pyodide.js` like:
```javascript
  let Module = {};
  Module._module = Module; // add this
  self.Module = Module;
```

**Comment by dalcde** [comment]
When setting up a test, we run `Error.stackTraceLimit = Infinity` via
run_js, when pyodide has not yet been set up.

**Comment by hoodmane** [comment]
@rth @dalcde The tests failed because now it correctly fails `test_numpy.py::test_python2js_numpy_dtype` which leaves the error indicator set.

**Comment by dalcde** [comment]
I can merge this after the tests from #1111 are tidied up

**Comment by hoodmane** [comment]
Okay, will do.

**Comment by hoodmane** [comment]
Oops should have merged first.

**Comment by hoodmane** [comment]
Also, the `array.array` test uses `Q` as the typecode. Javascript has a `BigUint64Array`. Ideally if I could make a `memoryview` with a definitely invalid format to make the test more forward compatible.

**Comment by dalcde** [comment]
For whatever inexplicable reason the tests seem to be genuinely failing

**Comment by hoodmane** [comment]
I will try to figure out what's wrong tomorrow.

**Comment by hoodmane** [comment]
I also added a second test with the `"u"` format, which is the only alternative to `Q` for a format string that can be made with `array.array` 

(`numpy` seems to make all sorts of other exotic format strings, from C I could make anything I want, but this seems good enough)

**Comment by hoodmane** [comment]
Now the docs is failing because `files.pythonhosted.org` is giving a 502...

**Pull request merged**